game_patch/object/entity.cpp

```diff
@@ -201,6 +202,79 @@ CodeInjection entity_process_pre_hide_riot_shield_injection{
     },
 };
 
+FunHook<void(int)> entity_blood_throw_gibs_hook{
+    0x0042E3C0, [](int handle) {
+        if (g_game_config.gibbing && !rf::is_multi) {
+            rf::Object* objp = rf::obj_from_handle(handle);
+
+            if (!objp) {
+                return; // invalid object
+            }
+
+            rf::Vector3 pos = objp->pos;
+            rf::GRoom* room = objp->room;
+            int explosion_vclip = 30;
+            int chunk_explosion_vclip = 30;
+            float explosion_vclip_radius = 1.0f;
+            rf::String debris_filename = "df_meatchunks0.V3D";
+
+            int debris_max_lifetime = 7000;
+            float debris_velocity = 8.5f;
+            float damage_scale = 1.0f;
+            rf::String cust_snd_set = "gib bounce";
+
+            if (objp->type == rf::OT_ENTITY) {
+                rf::Entity* ep = static_cast<rf::Entity*>(objp);
+
+                explosion_vclip = (ep->info->explode_vclip_index < 0)
+					? ep->info->explode_vclip_index : explosion_vclip;
+
+                explosion_vclip_radius = (ep->info->explode_vclip_radius > 0.0f)
+					? ep->info->explode_vclip_radius : explosion_vclip_radius;
+
+                debris_filename = (!ep->info->debris_filename.empty())
+					? ep->info->debris_filename : debris_filename;
+            }
+            else if (objp->type = rf::OT_CORPSE) {
```

Comment:
Not sure if corpses are handled properly. If you need `EntityInfo` to create a proper explosion, why not get it from corpse too? But i'm not even sure this code will be executed and when? Can you really damage corpses?

Reply:
Corpses do appear to be handled properly based on the code and observed behaviour in-game.

The corpse gibbing behaviour was actually already mostly present in `corpse_damage` - I barely had to edit it at all. Once I rebuilt `entity_blood_throw_gibs`, corpse gibbing was pretty much good to go, entity gibbing was what took way more work.

> Can you really damage corpses?

Yes, just in the stock game without this PR, they disappear from explosives instead of gibbing.